lemminx-liberty/src/main/java/io/openliberty/lemminx/liberty/services/FeatureService.java

```diff
@@ -46,41 +53,91 @@ private FeatureService() {
     String featureEndpoint = String.format(
         "https://repo1.maven.org/maven2/io/openliberty/features/features/%s/features-%s.json", libertyVersion,
         libertyVersion);
     InputStreamReader reader = new InputStreamReader(new URL(featureEndpoint).openStream());
+
+    // Only need the public features
+    ArrayList<Feature> publicFeatures = getPublicFeatureList(reader);
+
+    return publicFeatures;
+  }
+
+  /**
+   * Returns the default feature list
+   *
+   * @return list of features supported by the default version of liberty
+   */
+  private List<Feature> getDefaultFeatureList() {
+    try {
+      if (defaultFeatureList.isEmpty()) {
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        InputStream is = classloader.getResourceAsStream("features-20.0.0.9.json");
+        InputStreamReader reader = new InputStreamReader(is, StandardCharsets.UTF_8);
+
+        // Only need the public features
+        defaultFeatureList = getPublicFeatureList(reader);
+      }
+      return defaultFeatureList;
+
+    } catch (JsonParseException e) {
+      // unable to read json in resources file, return empty list
+      LOGGER.severe("Error: Unable to get default features.");
+      return defaultFeatureList;
+    }
+  }
+
+  /**
+   * Returns a list of public features
+   *
+   * @param reader - InputStreamReader for json feature list
+   * @return list of public features
+   */
+  private ArrayList<Feature> getPublicFeatureList(InputStreamReader reader) throws JsonParseException {
     Gson gson = new GsonBuilder().create();
     Feature[] featureList = new Gson().fromJson(reader, Feature[].class);
 
-    // Only need the public features
     ArrayList<Feature> publicFeatures = new ArrayList<>();
     Arrays.asList(featureList).stream()
-        .filter(f -> f.getWlpInformation().getVisibility().equals(LibertyConstants.PUBLIC_VISIBILITY))
-        .forEach(publicFeatures::add);
+            .filter(f -> f.getWlpInformation().getVisibility().equals(LibertyConstants.PUBLIC_VISIBILITY))
+            .forEach(publicFeatures::add);
+    defaultFeatureList = publicFeatures;
     return publicFeatures;
   }
 
-  public List<Feature> getFeatures(String libertyVersion) throws IOException {
+  public List<Feature> getFeatures(String libertyVersion) {
     LOGGER.fine("Getting features for version: " + libertyVersion);
     // if the features are already cached in the feature cache
     if (featureCache.containsKey(libertyVersion)) {
       return featureCache.get(libertyVersion);
     }
     // else need to fetch the features from maven central
-    List<Feature> features = fetchFeaturesForVersion(libertyVersion);
-    featureCache.put(libertyVersion, features);
-    return features;
+    try {
+      // verify that 2 min has gone by since last fetch request
+      long currentTime = System.currentTimeMillis();
+      if (currentTime >= (this.featureUpdateTime + 120000)) {
```

Comment:
Should this time duration be configurable by the user?

Reply:
we can add it as a setting similar to `libertyVersion`, @rzgry  what do you think?

Reply:
Yeah i think thats is a good idea.

Reply:
added `xml.liberty.requestDelay` as a configurable xml setting, accepts the number of seconds (> 0) between fetch request times